deps/chakrashim/core/lib/Jsrt/Jsrt.cpp

```diff
@@ -3433,6 +3433,178 @@ CHAKRA_API JsStringFree(_In_ char* stringValue)
     return JsNoError;
 }
 
+
+// -------- Experimental new String APIs ----------------------------------
+
+
+template <class SrcChar, class DstChar>
+static void CastCopy(const SrcChar* src, DstChar* dst, size_t count)
+{
+    const SrcChar* end = src + count;
+    while (src < end)
+    {
+        *dst++ = static_cast<DstChar>(*src++);
+    }
+}
+
+CHAKRA_API JsCreateString(
+    const char *content,
+    size_t length,
+    _Out_ JsValueRef *value)
+{
+    PARAM_NOT_NULL(content);
+
+    AutoArrayPtr<uint16_t> data(HeapNewNoThrowArray(uint16_t, length), length);
+    if (!data)
+    {
+        return JsErrorOutOfMemory;
+    }
+
+    // Cast source char to "unsigned" for correct cast
+    CastCopy((unsigned char*)content, (uint16_t*)data, length);
+
+    return JsPointerToString(
+        reinterpret_cast<const char16*>((uint16_t*)data), length, value);
+}
+
+CHAKRA_API JsCreateStringUtf8(
+    const uint8_t *content,
+    size_t length,
+    _Out_ JsValueRef *value)
+{
+    PARAM_NOT_NULL(content);
+    utf8::NarrowToWide wstr((LPCSTR)content, length);
+    if (!wstr)
+    {
+        return JsErrorOutOfMemory;
+    }
+
+    return JsPointerToString(wstr, wstr.Length(), value);
+}
+
+CHAKRA_API JsCreateStringUtf16(
+    const uint16_t *content,
+    size_t length,
+    _Out_ JsValueRef *value)
+{
+    return JsPointerToString(
+        reinterpret_cast<const char16*>(content), length, value);
+}
+
+
+CHAKRA_API JsWriteString(
+    JsValueRef value,
+    char* buffer,
+    size_t bufferSize,
+    _Out_opt_ size_t* length)
+{
+    const char16* str = nullptr;
+    size_t strLength = 0;
+    JsErrorCode errorCode = JsStringToPointer(value, &str, &strLength);
+    if (errorCode != JsNoError)
+    {
+        return errorCode;
+    }
+
+    if (!buffer)
+    {
+        if (length)
+        {
+            *length = strLength;
+        }
+    }
+    else
+    {
+        size_t count = min(bufferSize, strLength);
+        CastCopy(str, buffer, count);
+        if (length)
+        {
+            *length = count;
+        }
+    }
+
+    return JsNoError;
+}
+
+CHAKRA_API JsWriteStringUtf8(
```

Comment:
These names are unstable yet, could change in future. Feeling adding that is more clear but too long.